### ISSUE TITLE
Tensorboard hangs while loading Pytorch Profiler results.

### ISSUE THREAD

**Issue by bnsh** [pytorch/kineto#622]
Hi. This isn't a "issue" per se, mostly because I'm pretty sure I'm doing something wrong.

Here's my code snippet (minus the entire network that I was profiling in a train loop...

```
    with SummaryWriter(tb_dir) as writer, open(logfn, "wt", encoding="utf-8") as logfp, \
        torch.profiler.profile(
            activities=[
                torch.profiler.ProfilerActivity.CPU,
                torch.profiler.ProfilerActivity.CUDA,
            ],
            schedule=torch.profiler.schedule(
                wait=1,
                warmup=1,
                active=2
            ),
            on_trace_ready=torch.profiler.tensorboard_trace_handler(profile_dir, worker_name="hmm")
        ) as profiler:
        while (args.max_epochs is None or epoch < args.max_epochs) and (args.max_time is None or total_elapsed < args.max_time):
            train(model, epoch, train_dataset, ema_node_loss, opt, sched, crit) # This is a function defined elsewhere that just does the training loop.
            # a whole bunch of other stuff deleted for brevity
            profiler.step()
```

And, it _does_ write into my profile_dir these rather ginormous json files:

```
-rw-rw-r-- 1 binesh binesh 10049360679 Jun 20 19:40 hmm.1655768373559.pt.trace.json
-rw-rw-r-- 1 binesh binesh 10094463514 Jun 20 20:54 hmm.1655772846630.pt.trace.json
```

Then, after installing the [tensorboard plugin](https://github.com/pytorch/kineto/tree/master/tb_plugin), I go to http://localhost:6006/ and I see it just constantly "loading"... I've left it running for more than an hour to no avail.

I'm _pretty_ sure this isn't the place for this, since it's not really a "issue", but I didn't know if there was a gitter or a slack or something for the pytorch profiler. If there is such a thing, then please point me to that, and I'll ask my question there...

Thanks!
Binesh

**Comment by bnsh** [comment]
Hi @aaronenyeshi Thanks for your response.. OK, so I'm trying that, and I got a file that's half the size:

```
-rw-rw-r-- 1 binesh binesh 5021872278 Jun 22 03:37 hmm.1655883440285.pt.trace.json
```

But, it still (at least so far...) is just hanging. I'll let it keep running for a few more hours...

FWIW, this is what I see... Along with the Chrome developer tools/network trace showing that it keeps asking for "run" from the server, which keeps just returning

```
{"runs": [], "loading": true}
```
![loading](https://user-images.githubusercontent.com/353681/174975363-d7f14714-41b0-4de3-abca-e122831f6fd9.png)

**Comment by bnsh** [comment]
(FWIW: For my purposes, while it's interesting, and probably would be cool to get it working with Tensorboard, if there's some alternate way of looking at the profile data, I'd be OK with that as well..)

**Comment by bnsh** [comment]
Oh! It finally loaded! I wonder why it took so long before... Is it just the "repeat=1" that did it?

![loaded](https://user-images.githubusercontent.com/353681/174977794-c6ac4b2a-2cb8-43fb-926e-0351604032af.png)

**Comment by bnsh** [comment]
> It might help to run py-spy (https://github.com/benfred/py-spy) to get a sense of where the loading is spending its time.

Oh, that's really cool.. But, I guess I'd have to run that on tensorboard, right?

**Comment by aaronenyeshi** [comment]
> Oh! It finally loaded! I wonder why it took so long before... Is it just the "repeat=1" that did it?

Using `repeat=1` will make the profiler run once. If unset, it will run warmup + active in a cycle indefinitely. That may slow down your workload since its constantly profiling plus writing the trace to disk.
